frontend/src/app/services/auth/auth.service.ts

```diff
@@ -36,20 +64,46 @@ export class AuthService {
     );
   }
 
-  refreshToken(refresh_token: string): Observable<RefreshTokenResponse> {
-    return this.http.put<PostTokenResponse>(
-      environment.backend_url + '/authentication/tokens',
-      { refresh_token }
-    );
+  performTokenRefresh(): Observable<RefreshTokenResponse> {
+    const refreshToken = this._refreshToken;
+    return this.http
+      .put<PostTokenResponse>(
+        environment.backend_url + '/authentication/tokens',
+        { refreshToken }
+      )
+      .pipe(
+        map((res) => {
+          this.logIn(res.access_token, res.refresh_token);
+          return res;
+        }, first())
+      );
   }
 
   isLoggedIn(): boolean {
-    return !!this.localStorageService.getValue('access_token');
+    return !!this._accessToken;
+  }
+
+  logIn(accessToken: string, refreshToken: string) {
+    this._accessToken = accessToken;
+    this._refreshToken = refreshToken;
+    if (!environment.production) {
+      this.localStorageService.setValue('access_token', accessToken);
+      this.localStorageService.setValue('refresh_token', refreshToken);
+    }
```

Comment:
Is there a reason of still using `localStorageService` in the local environment? I would also remove that for the consistency.